tools/CoqMakefile.in

```diff
@@ -380,9 +380,14 @@ optfiles: $(if $(DO_NATDYNLINK),$(CMXSFILES))
 .PHONY: optfiles
 
 # FIXME, see Ralph's bugreport
+quick: COQQUICK := -quick
 quick: $(VOFILES:.vo=.vio)
 .PHONY: quick
 
+quick-light: COQQUICK := -quick-light
+quick-light: $(VOFILES:.vo=.vio)
+.PHONY: quick
```

Comment:
Wrong .PHONY

Reply:
- I fear that it would be reduce the usability of the system to reuse the same extension .vio. Depending on whether -quick or -quick-light is used, the Makefile rules for building .vo files would be different, this is unsatisfying. Why not introduce a ".voi" extension, describing "compiled interface". Remark: in the future, the implementation of the code that produce ".voi" might be completely different from that that produces ".vio" files, and in fact the latter might get dropped at some point. 

- In passing, I would recommand changing the name from "-quick-light" (which, out of context, is really not intuitive/informative), to "-output-voi" (or possibly just "-voi"). If the review process had existed in the past, I would have suggested "-quick" to be named "-output-vio", much to better reflect the intention. For comparison, consider the example: "latexmk -pdf foo.tex", to indicate pdf output.)

- It seems that there is a missing option, for indicating Coq whether it should load .vo or .vio files of dependencies. This is very important. The current behavior (based on timestamp) is inadequate, and makes it very hard for "make" to work properly. I thus suggest the coqc option "-load-voi" to indicate that coqc should load .voi and not .vo files for dependency. It should fail if it does not find .voi. It might appear as a best effort to load the .vo if it exists, but if the .vo is not sufficiently up to date then this is doing more bad than good. Btw, using a different extension (voi vs vio) would also avoid changing the current behavior of .vio files, and avoid breaking the existing workflow.

- It is useful to make a distinction between "a .v file whose proofs have been checked against .vo dependencies" (producing a .vo file) and "a .v file whose proofs have been checked against .voi dependencies". The later should produce a ".vok" file, with empty contents, simply to indicate that the file has been successfully checked. The reason is that in the former case, universes have not been checked. It is useful (to avoid complete recompilation of an entire project) to be able to make coexist .vo and .vok files. It is sane to never produce a .vo file that would lack universe checks. In summary, the "-load-voi" option should produce a ".vok" file and not a ".vo" file as output.

- In the code I read "| BuildVio of bool"
=> I think that it would be more explicit to replace the boolean by named constructors, e.g. | VioWithProofs | VioWithoutProofs
=> Yet given what is written above, and the fact that vio might become deprecated one day (the day OCaml comes with proper multicore support), I would suggest to simply have "| BuildVio | BuildVoi" in the main datatype.

- Bonus to your patch: extend Coq_makefile generator to produce dependency rules of the form: "Foo.requires: Bar.voi", when Foo.v requires Bar.v.

Thanks